src/bundle/DependencyInjection/Compiler/RememberMeServicesDecoratorCompilerPass.php

```diff
@@ -16,22 +16,28 @@
  */
 class RememberMeServicesDecoratorCompilerPass implements CompilerPassInterface
 {
-    private const REMEMBER_ME_LISTENER_ID_PREFIX = 'security.authentication.listener.rememberme.';
+    private const REMEMBER_ME_AUTHENTICATION_LISTENER_ID_PREFIX = 'security.authentication.listener.rememberme.';
+    private const REMEMBER_ME_AUTHENTICATOR_LISTENER_ID_PREFIX = 'security.listener.remember_me.';
 
     public function process(ContainerBuilder $container): void
     {
         // Find all remember-me listener definitions
         foreach ($container->getDefinitions() as $definitionId => $definition) {
-            if (0 === strpos($definitionId, self::REMEMBER_ME_LISTENER_ID_PREFIX)) {
-                $this->decorateRememberMeServices($container, $definition);
+            if (0 === strpos($definitionId, self::REMEMBER_ME_AUTHENTICATION_LISTENER_ID_PREFIX)) {
+                $this->decorateRememberMeServices($container, $definition, 1);
+            }
+
+            // New security authenticator system
+            if (0 === strpos($definitionId, self::REMEMBER_ME_AUTHENTICATOR_LISTENER_ID_PREFIX)) {
```

Comment:
This is quite ugly. With authenticators I could also decorate the listener itself to prevent the remember-me cookie from being set.